src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
here the logic is working wrong. If `onlyWithSon` param is true you have to use another filter. I would suggest preparing a list of men to use further (as 'sons' for filtering).
```suggestion
  const sons = people.filter(person => person.sex === 'm');
  const children = onlyWithSon
    ? people.filter((person) => sons.find(son => son.mather === person.name || son.father === person.name))
    : people;
```